sycl/source/detail/platform_util.cpp

```diff
@@ -62,21 +72,49 @@ uint32_t PlatformUtil::getMaxClockFrequency() {
   Buff = Buff.substr(Buff.rfind(' '), Buff.length());
   Freq *= std::stod(Buff);
   return Freq;
+#else
+#warning Your platform is not supported!
+#endif
+  return 0;
 }
 
 uint32_t PlatformUtil::getMemCacheLineSize() {
+#if defined(__x86_64__) || defined(__i386__)
   uint32_t CPUInfo[4];
   cpuid(CPUInfo, 0x80000006);
   return CPUInfo[2] & 0xff;
+#elif defined(SYCL_RT_OS_LINUX) && defined(_SC_LEVEL2_DCACHE_LINESIZE)
+  long linesize = sysconf(_SC_LEVEL2_DCACHE_LINESIZE);
```

Comment:
NIT: could you use camel case for variable names like in the surrounding code.

Reply:
Okay; clang-format didn't care so I didn't know.